pkg/resource/deploy/builtins.go

```diff
@@ -286,7 +286,13 @@ func (p *builtinProvider) readStackReference(inputs resource.PropertyMap) (resou
 
 	outputs, err := p.backendClient.GetStackOutputs(p.context, name.StringValue())
 	if err != nil {
-		return nil, err
+		var decryptionErr *StackOutputDecryptionError
+		if !errors.As(err, &decryptionErr) {
+			return nil, err
+		}
+
+		p.diag.Infof(diag.Message("", "eliding undecryptable secrets for stack reference '%s'"), name.StringValue())
```

Comment:
wonder if we should list which property paths where elided? "eliding undecryptable secrets 'foo', 'bar.baz', for stack reference 'a/b/c'"?

Reply:
Discussed and we don't think this is possible right now so we have gone with the existing mechanism.